server/api/api/utils.py

```diff
@@ -64,6 +65,18 @@ def log_and_raise(status=HTTPStatus.BAD_REQUEST.value, **kw):
     raise HTTPException(status_code=status, detail=kw)
 
 
+@functools.lru_cache
```

Comment:
I would add a relatively small `maxsize` here (16 or something like that), otherwise the cache would inflate to 1440 entities (# of minutes in a day), due to using `ttl_seconds` as parameter, each holding a full `ClientSpec` in it.
Also, it's confusing to call the param `ttl_seconds` since it actually represents the minutes, not how many seconds to live. As a matter of fact, each cached value will last a full day given the current implementation.